src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -248,7 +245,8 @@ public TopicManagementService(Map<String, Object> props, String serviceName) {
     _executor = Executors.newSingleThreadScheduledExecutor(new TopicManagementServiceThreadFactory());
     _configuredTopicReplicationFactor = config.getInt(TopicManagementServiceConfig.TOPIC_REPLICATION_FACTOR_CONFIG);
     _topicCreationEnabled = config.getBoolean(TopicManagementServiceConfig.TOPIC_CREATION_ENABLED_CONFIG);
-    _topicFactoryConfig = config.getString(TopicManagementServiceConfig.TOPIC_FACTORY_CONFIG);
+    String topicFactoryConfig = config.getString(TopicManagementServiceConfig.TOPIC_FACTORY_CONFIG);
+    _topicFactory = (TopicFactory) Class.forName(topicFactoryConfig).getConstructor(Map.class).newInstance(props);
```

Comment:
`props` is the properties of `TopicManagementService`. If user were to pass any config for the factory class, it should probably be specified as an additional map inside `props`. How about we add a config under `TopicManagementServiceConfig`, nam `topic-management.topicFactory.props`, and use its value to construct TopicFactory?